# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsPageContentFrame.cpp
# Commit: a2da895a58ce
# Full Hash: a2da895a58ce094d55e6a2e040dd648962b39868
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2022-02-22 09:37:09
# Regressor Bug: 1714584
# File Overlap Count: 2
# Description:
#   Bug 1714584 - Part 1: Decouple nsDisplayList internal list from nsDisplayItems r=mstange
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138152
# ==============================================================================

diff -r c3b3315d2344 -r a2da895a58ce layout/generic/nsPageContentFrame.cpp
--- a/layout/generic/nsPageContentFrame.cpp	Mon Feb 21 21:57:03 2022 +0000
+++ b/layout/generic/nsPageContentFrame.cpp	Tue Feb 22 00:44:25 2022 +0000
@@ -301,7 +301,7 @@
   if (!aExtraPage->HasAnyStateBits(NS_FRAME_FORCE_DISPLAY_LIST_DESCEND_INTO)) {
     return;
   }
-  nsDisplayList list;
+  nsDisplayList list(aBuilder);
   aExtraPage->BuildDisplayListForStackingContext(aBuilder, &list);
   PruneDisplayListForExtraPage(aBuilder, aPage, &list);
   aList->AppendToTop(&list);
@@ -332,7 +332,7 @@
 
   nsDisplayListCollection set(aBuilder);
 
-  nsDisplayList content;
+  nsDisplayList content(aBuilder);
   {
     const nsRect clipRect(aBuilder->ToReferenceFrame(this), GetSize());
     DisplayListClipState::AutoSaveRestore clipState(aBuilder);